lib/route/src/Obelisk/Route.hs

```diff
@@ -692,7 +694,9 @@ shadowEncoder f g = Encoder $ do
   overlaps <- fmap catMaybes $ forM universe $ \a -> do
     let c = _encoderImpl_encode vf a
     mb <- gCanParse c
-    pure $ fmap (\b -> (a, b, c)) mb
+    pure $ mb >>= \b -> do
+      guard $ _encoderImpl_encode vg b == c
```

Comment:
This will stop working when #1075 is merged, but if I add `import Control.Monad` here the `-Werror` will cause this to fail.

Reply:
We could/should disable -Werror for redundant imports. Ideally it would be an error to have imports that are redundant for *all supported versions*, but I'm not sure to achieve that.